modules/rostests/apitests/ntdll/NtSetInformationProcess.c

```diff
@@ -7,6 +7,140 @@
 
 #include "precomp.h"
 
+static
+void
+Test_ProcBasePriorityClass(void)
+{
+    NTSTATUS Status;
+
+    /*
+     * Assign a priority of HIGH_PRIORITY (see pstypes.h).
+     * The function will fail with a status of STATUS_PRIVILEGE_NOT_HELD
+     * as the process who executed the caller doesn't have the requested
+     * privileges to perform the operation.
+    */
+    KPRIORITY BasePriority = HIGH_PRIORITY;
+
+    /* Everything is NULL */
+    Status = NtSetInformationProcess(NULL,
+                                     ProcessBasePriority,
+                                     NULL,
+                                     0);
+    ok_hex(Status, STATUS_INFO_LENGTH_MISMATCH);
+
+    /* Set the base priority to an invalid process handle */
+    Status = NtSetInformationProcess(NULL,
+                                     ProcessBasePriority,
+                                     &BasePriority,
+                                     sizeof(KPRIORITY));
+    ok_hex(Status, STATUS_INVALID_HANDLE);
+
+    /* Don't input anything to the caller but the length information is valid */
+    Status = NtSetInformationProcess(NtCurrentProcess(),
+                                     ProcessBasePriority,
+                                     NULL,
+                                     sizeof(KPRIORITY));
+    ok_hex(Status, STATUS_ACCESS_VIOLATION);
+
+    /* Give the base priority input to the caller but length is invalid */
+    Status = NtSetInformationProcess(NtCurrentProcess(),
+                                     ProcessBasePriority,
+                                     &BasePriority,
+                                     0);
+    ok_hex(Status, STATUS_INFO_LENGTH_MISMATCH);
+
+    /* The input buffer is misaligned and length information invalid */
+    Status = NtSetInformationProcess(NtCurrentProcess(),
+                                     ProcessBasePriority,
+                                     (PVOID)1,
+                                     0);
+    ok_hex(Status, STATUS_INFO_LENGTH_MISMATCH);
+
+    /* The input buffer is misaligned */
+    Status = NtSetInformationProcess(NtCurrentProcess(),
+                                     ProcessBasePriority,
+                                     (PVOID)1,
```

Comment:
For completeness, test `2` as well to show that the alignment is indeed four and not two (which is unlikely, but still...)